src/main/java/com/rebuy/consul/ConsulService.java

```diff
@@ -0,0 +1,38 @@
+package com.rebuy.consul;
+
+import com.ecwid.consul.v1.ConsulClient;
+import com.ecwid.consul.v1.agent.model.NewService;
+
+public class ConsulService
+{
+    public static ConsulServiceBuilder builder()
+    {
+        return new ConsulServiceBuilder();
+    }
+
+    final ConsulClient client;
+
+    final NewService service;
+
+    public ConsulService(ConsulClient client, NewService service)
+    {
+        this.client = client;
+        this.service = service;
+    }
+
+    public void register()
+    {
+        client.agentServiceRegister(service);
+    }
+
+    public void shutdownHook()
+    {
+        Runtime.getRuntime().addShutdownHook(
+            new Thread(this::unregister));
```

Comment:
That's how the Java API works: https://docs.oracle.com/javase/8/docs/api/java/lang/Runtime.html#addShutdownHook-java.lang.Thread-